proto/spire/plugin/server/credentialcomposer/v1/credentialcomposer.proto

```diff
@@ -0,0 +1,179 @@
+// The CredentialComposer plugin influences the attributes of X.509
+// certificates and JWT tokens minted by or on behalf of SPIRE Server.
+
+syntax = "proto3";
+package spire.plugin.server.credentialcomposer.v1;
+option go_package = "github.com/spiffe/spire-plugin-sdk/proto/spire/plugin/server/credentialcomposer/v1;credentialcomposerv1";
+
+import "google/protobuf/value.proto";
+
+service CredentialComposer {
+    // Composes the SPIRE Server X509 CA.
+    rpc ComposeServerX509CA(ComposeServerX509CARequest) returns (ComposeServerX509CAResponse);
+
+    // Composes the SPIRE Server X509-SVID.
+    rpc ComposeServerX509SVID(ComposeServerX509SVIDRequest) returns (ComposeServerX509SVIDResponse);
+
+    // Composes the SPIRE Agent X509-SVID.
+    rpc ComposeAgentX509SVID(ComposeAgentX509SVIDRequest) returns (ComposeAgentX509SVIDResponse);
+
+    // Composes workload X509-SVIDs.
+    rpc ComposeWorkloadX509SVID(ComposeWorkloadX509SVIDRequest) returns (ComposeWorkloadX509SVIDResponse);
+
+    // Composes workload JWT-SVIDs.
+    rpc ComposeWorkloadJWTSVID(ComposeWorkloadJWTSVIDRequest) returns (ComposeWorkloadJWTSVIDResponse);
+}
+
+message ComposeServerX509CARequest {
+    // The default attributes for the server X509 CA.
+    X509CAAttributes attributes = 1;
+}
+
+message ComposeServerX509CAResponse {
+    // The attributes for the X509 CA. If not included in the response, the
+    // default attributes sent in the request will be used.
+    X509CAAttributes attributes = 1;
+}
+
+message ComposeServerX509SVIDRequest {
+    // The default attributes for the server X509-SVID.
+    X509SVIDAttributes attributes = 1;
+}
+
+message ComposeServerX509SVIDResponse {
+    // The attributes for the X509-SVID. If not included in the response, the
+    // default attributes sent in the request will be used.
+    X509SVIDAttributes attributes = 1;
+}
+
+message ComposeAgentX509SVIDRequest {
+    // The default attributes for the server X509-SVID.
+    X509SVIDAttributes attributes = 1;
+}
+
+message ComposeAgentX509SVIDResponse {
+    // The attributes for the X509-SVID. If not included in the response, the
+    // default attributes sent in the request will be used.
+    X509SVIDAttributes attributes = 1;
+}
+
+message ComposeWorkloadX509SVIDRequest {
+    // The default attributes for the X509-SVID.
+    X509SVIDAttributes attributes = 1;
+
+    // The SPIFFE ID of the workload.
+    string spiffe_id = 2;
+
+    // PKIX encoded public key of the workload.
+    bytes public_key = 3;
```

Comment:
what uses has to have the public_key exposed to plugin?

Reply:
One use case was somebody using this plugin to reject agent signing requests for public keys that are disallowed by their infrastructure (e.g. weak keys, EC v.s. RSA). Right now the server does not care about the agent key at all.